It is not working. Here the full log:

 Action: fixServerRole InstanceId: i-0dfb47f1c2f664107
   ------- describe_iam_profile: i-0dfb47f1c2f664107 - associated
  Instance IAM Profile is invalid: arn:aws:iam::514046899996:instance-profile/Admin-Ec2
  Disassociating IAM profile: iip-assoc-0facf7cd3440a7f30
   ------- describe_iam_profile: i-0dfb47f1c2f664107 - disassociated
   Error checking disassociation status: An error occurred (InvalidAssociationID.NotFound) when calling the DescribeIamInstanceProfileAssociations operation: An invalid association-id of 'iip-assoc-0facf7cd3440a7f30' was given. Assuming profile is disassociated.
 Attaching IAM role to the Minecraft Instance
 Attaching IAM profile: AWSEC2MinecraftProfile-us-west-2
  LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
  [ERROR] AttributeError: <botocore.errorfactory.EC2Exceptions object at 0x7fe45c2cd940> object has no attribute UnauthorizedOperation. Valid exceptions are: Traceback (most recent call last):   File "/var/task/index.py", line 404, in handler     return action_process(field_name,instance_id,input)   File "/var/task/index.py", line 215, in action_process     return handler()   File "/var/task/index.py", line 204, in <lambda>     "fixserverrole": lambda: handle_fix_role(instance_id),   File "/var/task/index.py", line 278, in handle_fix_role     resp = iam_profile.manage_iam_profile()   File "/var/task/index.py", line 49, in manage_iam_profile     return self.attach_iam_profile()   File "/var/task/index.py", line 99, in attach_iam_profile     except self.ec2_client.exceptions.UnauthorizedOperation as e:   File "/opt/python/botocore/errorfactory.py", line 51, in __getattr__     raise AttributeError(

--- a/lambdas/serverAction/index.py
+++ b/lambdas/serverAction/index.py
@@ -6,6 +6,7 @@ import time
 import authHelper
 import ec2Helper
 import utilHelper
+import botocore.exceptions
 
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
@@ -56,11 +57,15 @@ class IamProfile:
         try:
             # Call EC2 API to remove the profile association
             self.ec2_client.disassociate_iam_instance_profile(AssociationId=self.association_id)
-        except self.ec2_client.exceptions.InvalidAssociationIDNotFound as e:
-            # Handle the case where the association ID doesn't exist
-            logger.warning("Association ID not found: %s. This is not an error, continuing...", self.association_id)
-            # Return True to continue with attaching the profile
-            return True
+        except botocore.exceptions.ClientError as e:
+            if "InvalidAssociationID.NotFound" in str(e):
+                # Handle the case where the association ID doesn't exist
+                logger.warning("Association ID not found: %s. This is not an error, continuing...", self.association_id)
+                # Return True to continue with attaching the profile
+                return True
+            else:
+                logger.error("Error disassociating IAM profile: %s", str(e))
+                return False
         except Exception as e:
             logger.error("Error disassociating IAM profile: %s", str(e))
             return False
@@ -96,11 +101,15 @@ class IamProfile:
                 IamInstanceProfile={"Name": ec2_instance_profile_name},
                 InstanceId=self.instance_id
             )
-        except self.ec2_client.exceptions.UnauthorizedOperation as e:
-            # Handle unauthorized operation specifically
+        except botocore.exceptions.ClientError as e:
             error_msg = str(e)
-            logger.error("Unauthorized operation when attaching IAM profile: %s", error_msg)
-            return {"status": "error", "message": error_msg, "code": "UnauthorizedOperation"}
+            # Check if this is an unauthorized operation
+            if "UnauthorizedOperation" in error_msg:
+                logger.error("Unauthorized operation when attaching IAM profile: %s", error_msg)
+                return {"status": "error", "message": error_msg, "code": "UnauthorizedOperation"}
+            else:
+                logger.error("Error attaching IAM profile: %s", error_msg)
+                return False
         except Exception as e:
             logger.error("Error attaching IAM profile: %s", str(e))
             return False

